src/store/modules/delegates/actions.js

```diff
@@ -54,19 +54,32 @@ export default {
       context.commit('reset')
     }
   },
-  getDelegates (context, payload) {
-    admApi.getDelegatesWithVotes(payload.address).then(response => {
-      if (response.success) {
-        const votes = response.delegates.map(vote => vote.address)
-        admApi.getDelegatesCount().then(response => {
-          if (response.success) {
-            for (let i = 0; i < response.count / constants.Delegates.ACTIVE_DELEGATES; i++) {
-              _getDelegates(context, constants.Delegates.ACTIVE_DELEGATES, i * constants.Delegates.ACTIVE_DELEGATES, votes)
-            }
-          }
-        })
+  async getDelegates(context, payload) {
+    let votes = []
```

Comment:
why? variable declaration cannot throw an error

Reply:
For readability. The variable `votes` is not changed nor used in the outer scope. Since `votes` can be changed only once, you can just put its definition just before `if` statement where it changes

```js
const votesResponse = await admApi.getDelegatesWithVotes(payload.address)

let votes = []

if (votesResponse.success) {
  votes = votesResponse.delegates.map((vote) => vote.address)
}
```

or even use a ternary operator to prove that this variable will never change again:

```js
const votesResponse = await admApi.getDelegatesWithVotes(payload.address)

const votes = votesResponse.success
  ? votesResponse.delegates.map((vote) => vote.address)
  : []
```

Reply:
The second variant with ternary operator looks good.
Pushed the changes.